### PR TITLE
Propagate maxSegmentsPerCell and maxSynapsesPerSegment settings from OPF model params to Connections

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2738.

Depends on https://github.com/numenta/nupic.core/pull/687.

### CODE REVIEW

**Comment by chetan51** [comment]
@sco

File: src/nupic/research/fast_temporal_memory.py [hunk 1]
```diff
@@ -37,7 +37,10 @@ class FastTemporalMemory(TemporalMemory):
 
   def __init__(self, *args, **kwargs):
     super(FastTemporalMemory, self).__init__(*args, **kwargs)
-    self.connections = Connections(self.numberOfCells())
+    self.connections = Connections(
+      self.numberOfCells(),
+      maxSegmentsPerCell=kwargs["maxSegmentsPerCell"],
```
Comment (scottpurdy) [inline src/nupic/research/fast_temporal_memory.py]: Make this a keyword arg before kwargs.

**Comment by scottpurdy** [comment]
:+1: other than the one comment

**Comment by chetan51** [comment]
@scottpurdy I couldn't figure out how to add additional kwargs while still using `**kwargs`, so I just did this:
https://github.com/numenta/nupic/commit/282627b126d9cfdd16e288e1c0a5eb11728259f3

Is that okay?

File: src/nupic/research/fast_temporal_memory.py [hunk 2]
```diff
@@ -36,8 +36,14 @@ class FastTemporalMemory(TemporalMemory):
   """
 
   def __init__(self, *args, **kwargs):
+    maxSegmentsPerCell = kwargs.get("maxSegmentsPerCell") or 255
```
Comment (scottpurdy) [inline src/nupic/research/fast_temporal_memory.py]: `get` has an optional second parameter that is the default if the key isn't found.

Comment (chetan51) [inline src/nupic/research/fast_temporal_memory.py]: Okay, thanks.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**